wasm/src/secrets/secrets_resolver.rs

```diff
@@ -56,33 +94,28 @@ impl _SecretsResolver for JsSecretsResolver {
         Ok(secret)
     }
 
-    /// Find all secrets that have one of the given IDs.
-    /// Return secrets only for key IDs for which a secret is present.
-    ///
-    /// # Parameters
-    /// - `secret_ids` the IDs find secrets for
-    ///
-    /// # Returns
-    /// possible empty list of all secrets that have one of the given IDs.
     async fn find_secrets<'a>(&self, secret_ids: &'a [&'a str]) -> _Result<Vec<&'a str>> {
         let _secret_ids = secret_ids
             .into_iter()
             .map(|s| JsValue::from_str(s))
             .collect::<Array>();
 
         // TODO: better error conversion
-        let found = self.0.find_secrets(_secret_ids).await.map_err(|e| {
-            err_msg(
-                ErrorKind::InvalidState,
-                format!("Unable find secrets {:#?}", e),
-            )
-        })?;
+        let found = self
+            .0
+            .find_secrets(_secret_ids)
+            .await
+            .from_js()
```

Comment:
Should we make a context string a parameter for `from_js`?

Reply:
> Should we make a context string a parameter for from_js?

We don't need context param in `from_js` (keep single responsibility).  If context is required it can be added as next call

```
.from_js()
.context("...")?
```

In this case i think we don't have extra context. It i just thin wrapper over javascript function.